src/hooks/useSubmit/useSubmit.ts

```diff
@@ -219,18 +236,70 @@ const useSubmitWithSign = <ResponseType>(
   })
 }
 
-export type SignProps = {
-  publicClient: PublicClient
-  walletClient: WalletClient
+type SignBaseProps = {
   address: `0x${string}`
   payload: string
-  chainId: string
   forcePrompt: boolean
   keyPair?: CryptoKeyPair
   msg?: string
   ts: number
 }
 
+export type SignProps = SignBaseProps & {
+  publicClient: PublicClient
+  walletClient: WalletClient
+  chainId: string
+}
+
+export type FuelSignProps = SignBaseProps & { wallet: WalletUnlocked }
+
+const createMessageParams = (
+  address: `0x${string}`,
+  ts: number,
+  msg: string,
+  payload: string
+): MessageParams => ({
+  addr: address.toLowerCase(),
+  nonce: randomBytes(32).toString("base64"),
+  ts: ts.toString(),
+  hash: payload !== "{}" ? keccak256(stringToBytes(payload)) : undefined,
+  method: null,
+  msg,
+  chainId: undefined,
+})
+
+const signWithKeyPair = (keyPair: CryptoKeyPair, params: MessageParams) =>
+  window.crypto.subtle
+    .sign(
+      { name: "ECDSA", hash: "SHA-512" },
+      keyPair.privateKey,
+      Buffer.from(getMessage(params))
+    )
+    .then((signatureBuffer) => Buffer.from(signatureBuffer).toString("hex"))
+
+export const fuelSign = async ({
+  wallet,
+  address,
+  payload,
+  keyPair,
+  forcePrompt,
+  msg = DEFAULT_MESSAGE,
+  ts,
+}: FuelSignProps): Promise<[string, Validation]> => {
+  const params = createMessageParams(address, ts, msg, payload)
+  let sig = null
+
+  if (!!keyPair && !forcePrompt) {
+    params.method = ValidationMethod.KEYPAIR
+    sig = await signWithKeyPair(keyPair, params)
+  } else {
+    params.method = ValidationMethod.STANDARD
+    sig = await wallet.signMessage(getMessage(params))
+  }
```

Comment:
Hmmm, on the BE there is a new auth method, but here the same one seems to be used as for an EVM wallet :thinking:
The new one doesn't even seem to be defined on the enum, maybe the new method is completely unnecessary, and we only need the validator changes to accept a Fuel address on this method?

cc @marclllaks

Reply:
Very nice catch, we'll use the `FUEL` auth method from now: https://github.com/agoraxyz/guild.xyz/pull/950/commits/a8e03d8ff5009db3b52058984fd3a99d3385f568